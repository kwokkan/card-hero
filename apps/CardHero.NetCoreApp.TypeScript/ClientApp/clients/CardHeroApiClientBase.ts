// no export as this is used by NSwag generated api clients
class CardHeroApiClientBase {
    getBaseUrl(defaultBaseUrl: string, baseUrl: string): string {
        if (!baseUrl) {
            return defaultBaseUrl;
        }

        if (baseUrl.endsWith('/')) {
            // remove trailing slash as NSwag uses '/' as part of the api url
            return baseUrl.substring(0, baseUrl.length - 2);
        }

        return baseUrl;
    }
}

module.exports = CardHeroApiClientBase;