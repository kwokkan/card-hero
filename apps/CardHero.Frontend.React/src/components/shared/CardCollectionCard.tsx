import React from "react";
import { ICardCollectionModel, Rarity } from "../../clients/clients";
import { Icon } from "../../styles";

interface ICardCollectionCardProps {
    card: ICardCollectionModel;
    actionName?: string;
    onActionClicked?: (card: ICardCollectionModel) => void;
    actionDisabled?: boolean;
    actionClassName?: string;
}

export function CardCollectionCard(props: ICardCollectionCardProps) {
    const card = props.card;

    const onActionClicked = () => {
        if (props.onActionClicked) {
            props.onActionClicked(props.card);
        }
    };

    const actionClassName = "btn btn-sm " + (props.actionClassName || "");

    return (
        <div className="list-group-item d-flex justify-content-between align-items-center" data-card-collection-id={card.id}>
            <span>
                {card.card.name}
                {" "}
                <span className={"text-rarity_" + Rarity[card.card.rarity].toLowerCase()} title={Rarity[card.card.rarity]}>
                    {Array(card.card.rarity).fill(0).map((_, i) =>
                        <Icon key={i} icon="star" />
                    )}
                </span>
            </span>

            {props.actionName &&
                <button type="button" className={actionClassName} disabled={props.actionDisabled} onClick={onActionClicked}>
                    {props.actionName}
                </button>
            }
        </div>
    );
}
