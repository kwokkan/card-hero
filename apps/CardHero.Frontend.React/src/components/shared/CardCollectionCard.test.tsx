import React from "react";
import renderer from "react-test-renderer";
import { CardModel, ICardCollectionModel, ICardModel } from "../../clients/clients";
import { CardCollectionCard } from "./CardCollectionCard";

test("<CardCollectionCard /> renders empty card", () => {
    const data: ICardModel = {
        id: 1,
        name: "Test card"
    };
    const card: ICardCollectionModel = {
        card: CardModel.fromJS(data)
    };
    const tree = renderer
        .create(<CardCollectionCard card={card} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionCard /> renders card with custom actionName", () => {
    const data: ICardModel = {
        id: 1,
        name: "Test card"
    };
    const card: ICardCollectionModel = {
        card: CardModel.fromJS(data)
    };
    const tree = renderer
        .create(<CardCollectionCard card={card} actionName="Click me" />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionCard /> renders card with custom actionClassName", () => {
    const data: ICardModel = {
        id: 1,
        name: "Test card"
    };
    const card: ICardCollectionModel = {
        card: CardModel.fromJS(data)
    };
    const tree = renderer
        .create(<CardCollectionCard card={card} actionName="Click me" actionClassName="my-action" />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<CardCollectionCard /> renders card with disabled", () => {
    const data: ICardModel = {
        id: 1,
        name: "Test card"
    };
    const card: ICardCollectionModel = {
        card: CardModel.fromJS(data)
    };
    const tree = renderer
        .create(<CardCollectionCard card={card} actionName="Click me" actionDisabled={true} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
