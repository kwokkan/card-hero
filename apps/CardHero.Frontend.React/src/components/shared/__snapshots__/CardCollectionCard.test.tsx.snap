// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCollectionCard /> renders card with custom actionClassName 1`] = `
<div
  className="list-group-item d-flex justify-content-between align-items-center"
>
  Test card
  <button
    className="btn btn-sm my-action"
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`<CardCollectionCard /> renders card with custom actionName 1`] = `
<div
  className="list-group-item d-flex justify-content-between align-items-center"
>
  Test card
  <button
    className="btn btn-sm "
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`<CardCollectionCard /> renders card with disabled 1`] = `
<div
  className="list-group-item d-flex justify-content-between align-items-center"
>
  Test card
  <button
    className="btn btn-sm "
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`<CardCollectionCard /> renders empty card 1`] = `
<div
  className="list-group-item d-flex justify-content-between align-items-center"
>
  Test card
</div>
`;
